python/fp.py

```diff
@@ -17,15 +20,27 @@ def __add__(self, other: "Elem") -> "Elem":
     def __sub__(self, other: "Elem") -> "Elem":
         return Elem(sub(self.n, other.n))
 
-    def __pow__(self, other: "Elem") -> "Elem":
-        return Elem(pow(self.n, other.n))
+    def __pow__(self, other: "Elem | int") -> "Elem":
+        match other:
+            case int():
+                return Elem(pow(self.n, other))
+            case Elem():
+                return Elem(pow(self.n, other.n))
+            case _:
+                raise Exception("??")
```

Comment:
I doubt these lines are correct, for they are montgomery encoded, so `self.n` has $R$ attached to itself.